src/map/status_effect_container.cpp

```diff
@@ -447,6 +448,7 @@ void CStatusEffectContainer::RemoveStatusEffect(uint32 id, bool silent)
         PChar->PLatentEffectContainer->CheckLatentsFoodEffect();
         PChar->PLatentEffectContainer->CheckLatentsStatusEffect();
         PChar->PLatentEffectContainer->CheckLatentsRollSong(HasStatusEffectByFlag(EFFECTFLAG_SONG | EFFECTFLAG_ROLL));
+        PChar->PLatentEffectContainer->CheckLatentsWeather();
```

Comment:
now the last thing: since weather can change via status effect, can you add the weather latent effect check into CheckLatentsStatusEffect instead of calling this?